### DISCUSSION TITLE
Adding Roles and Permissions to custom module

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16791] [Archive]
How do I add a manually created custom module to Roles and Permissions section and add its roles and permissions.
Is there any specific way to do it ? 
If there, any help will be appreciated.

Secondly, is there any way to create a custom module which exactly looks like module that are in-built?
I want to create a custom module exactly like this with same functionality : 
<img width="1440" alt="Screenshot 2022-12-12 at 10 48 37 AM" src="https://user-images.githubusercontent.com/117433805/206966428-7d697b49-4e9e-437e-9255-0addd7c39203.png">

**Comment by peauc** [comment]
Hello, please take a look at this discussion I started a few days ago, it has answers for your issues with a code exemple at the end, just remove the store call : https://github.com/directus/directus/discussions/16720.

You totally could copy the Directus UI but that will be a decent amount of custom code. I would recomand looking at the other type of extensions

**Reply by bhagya1101** [reply]
If I want to copy exact Directus UI where can I copy it from?

**Reply by peauc** [reply]
You can look at the sources but be warned that you will have to tinker a lot.

**Reply by bhagya1101** [reply]
Thanks a lot!!
